From a25104555d1f9e80c6cf04d4f3d74bd65cfef8aa Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Thu, 11 Aug 2011 09:29:04 +0300
Subject: [strace patch] fix data truncation (implicit 'printrusage32' definition)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Caught by gentoo's QA checks:
> alpha-unknown-linux-gnu-gcc -DHAVE_CONFIG_H -I.  -I./linux/alpha -I./linux -I./linux  -Wall -Wwrite-strings -mieee -pipe -O2 -m
> cpu=ev4 -MT process.o -MD -MP -MF .deps/process.Tpo -c -o process.o process.c
> process.c: In function ‘printwaitn’:
> process.c:1949: warning: implicit declaration of function ‘printrusage32’

 * QA Notice: Package has poor programming practices which may compile
 *            fine but exhibit random runtime failures.
 * process.c:1949: warning: implicit declaration of function ‘printrusage32’

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 defs.h |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/defs.h b/defs.h
index b34f442..7bd45ac 100644
--- a/defs.h
+++ b/defs.h
@@ -581,6 +581,9 @@ extern void printfd(struct tcb *, int);
 extern void printsock(struct tcb *, long, int);
 extern void print_sock_optmgmt(struct tcb *, long, int);
 extern void printrusage(struct tcb *, long);
+#ifdef ALPHA
+extern void printrusage32(struct tcb *, long);
+#endif
 extern void printuid(const char *, unsigned long);
 extern int clearbpt(struct tcb *);
 /*
-- 
1.7.3.4

